src/main/kotlin/net/atos/zac/smartdocuments/templates/SmartDocumentsTemplateConverter.kt

```diff
@@ -3,49 +3,147 @@ package net.atos.zac.smartdocuments.templates
 import net.atos.client.smartdocuments.model.templates.SmartDocumentsResponseTemplate
 import net.atos.client.smartdocuments.model.templates.SmartDocumentsResponseTemplateGroup
 import net.atos.client.smartdocuments.model.templates.SmartDocumentsTemplatesResponse
+import net.atos.zac.smartdocuments.rest.RESTSmartDocumentsTemplate
+import net.atos.zac.smartdocuments.rest.RESTSmartDocumentsTemplateGroup
 import net.atos.zac.smartdocuments.templates.model.SmartDocumentsTemplate
 import net.atos.zac.smartdocuments.templates.model.SmartDocumentsTemplateGroup
+import net.atos.zac.zaaksturing.model.ZaakafhandelParameters
 import java.time.ZonedDateTime
 
 @Suppress("TooManyFunctions")
 object SmartDocumentsTemplateConverter {
 
-    fun SmartDocumentsTemplatesResponse.toModel(): Set<SmartDocumentsTemplateGroup> =
+    fun SmartDocumentsTemplatesResponse.toREST(): Set<RESTSmartDocumentsTemplateGroup> =
         this.documentsStructure.templatesStructure.templateGroups
-            .mapTo(mutableSetOf()) { convertTemplateGroupToModel(it, null) }
+            .mapTo(mutableSetOf()) { convertTemplateGroupResponseToREST(it) }
+
+    private fun convertTemplateGroupResponseToREST(
+        group: SmartDocumentsResponseTemplateGroup
+    ): RESTSmartDocumentsTemplateGroup {
+        val restGroup = createRESTTemplateGroup(group)
+
+        if (!group.templates.isNullOrEmpty()) {
```

Comment:
I think this is better achieved like so:
```kotlin
restGroup.templates = group.templates?.map { createRESTTemplate(it) }?.toSet()
```
Or is there a logic error?

Reply:
Test is ok, which means it's equivalent, so replaced it. Thanks!